src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/prison/AddressService.kt

```diff
@@ -159,12 +142,44 @@ class AddressService(
       dateOfBirth = addResidentRequest.dateOfBirth,
       age = addResidentRequest.age,
       isMainResident = addResidentRequest.isMainResident,
-      standardAddressCheckRequest = standardAddressCheckRequest,
+      standardAddressCheckRequest = addressCheckRequest,
     )
     resident = residentRepository.save(resident)
     return resident.toSummary()
   }
 
+  @Transactional
+  fun updateCaseAdminAdditionalInformation(
+    prisonNumber: String,
+    requestId: Long,
+    caseAdminInfoRequest: UpdateCaseAdminAdditionInfoRequest,
+  ) {
+    val curfewAddressCheckRequest = getCurfewAddressCheckRequest(requestId, prisonNumber)
+    curfewAddressCheckRequest.caAdditionalInfo = caseAdminInfoRequest.additionalInformation
+    curfewAddressCheckRequestRepository.save(curfewAddressCheckRequest)
+  }
+
+  private fun getCurfewAddressCheckRequest(requestId: Long, prisonNumber: String): CurfewAddressCheckRequest {
+    val curfewAddressCheckRequest =
+      curfewAddressCheckRequestRepository.findByIdOrNull(requestId)
+        ?: throw EntityNotFoundException("Cannot find standard address check request with id: $requestId")
```

Comment:
guess these messages aren't quite right as they could be any type of address check?